File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        #nums.sort()
        counter = [0, 0, 0]
        for color in nums:
            counter[color] += 1
        start = 0
        for i, cnt in enumerate(counter):
            for j in range(start, start+cnt):
                nums[j] = i
            start += cnt
        
        

File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n = nums.size();
        int right = n - 1; 
        for (int i = 0; i < n; i++) {
            if (nums[i] == 2) {
                while (nums[right] == 2 && i < right) {
                    right--;
                }
                swap(nums[i], nums[right]);
            }
        }
        for (int i = 0; i < right; i++) {
            if (nums[i] == 1) {
                while (nums[right] >= 1 && i < right) {
                    right--;
                }
                swap(nums[i], nums[right]);
            }
        }
    }
};